third_party/2and3/chardet/universaldetector.pyi

```diff
@@ -0,0 +1,19 @@
+from typing import Dict, Union, AnyStr, Pattern
+from logging import Logger
+
+class UniversalDetector:
+    MINIMUM_THRESHOLD: float
+    HIGH_BYTE_DETECTOR: Pattern[bytes]
+    ESC_DETECTOR: Pattern[bytes]
+    WIN_BYTE_DETECTOR: Pattern[bytes]
+    ISO_WIN_MAP: Dict[str, str]
+
+    result: Dict[str, Union[str, float]]
```

Comment:
You could use a `TypedDict` for the result and the return type of `close()`. Also please note that during processing, both the `encoding` and `language` attributes can be `None`, which should be reflected here (but not in the return type of `close()`).

Reply:
So shall we make two different `TypedDict`? One for the intermediate result and the other for the final value?

Reply:
I think that would make sense.